module/zfs/zfs_ioctl.c

```diff
@@ -1393,9 +1393,9 @@ get_zfs_sb(const char *dsname, zfs_sb_t **zsbp)
 
 	mutex_enter(&os->os_user_ptr_lock);
 	*zsbp = dmu_objset_get_user(os);
-	if (*zsbp && (*zsbp)->z_sb) {
-		atomic_inc(&((*zsbp)->z_sb->s_active));
-	} else {
+	/* bump s_active only when non-zero to prevent umount race */
+	if (!*zsbp || !(*zsbp)->z_sb ||
+	    !atomic_inc_not_zero(&((*zsbp)->z_sb->s_active))) {
```

Comment:
The fix here looks good but let's structure it as a `== NULL` check for readability.

`if (*zsbp == NULL || (*zsbp)->z_sb == NULL || ...`